### DISCUSSION TITLE
[Support]: Reporting of PTZ position

### DISCUSSION THREAD

**Discussion by jakeycrx** [blakeblackshear/frigate#12247] [General Support] [answered]
### Describe the problem you are having

Hi,

I've just set up my first Dahua PTZ camera with autotracking.  I want to set up a zone to trigger a security light at night, but I would like the automation to only fire when the PTZ is parked in the home position. I'm thinking it will probably incorrectly trigger the light when autotracking.

I have checked the Frigate, Dahua and ONVIF integrations, but I can't find any mention of the PTZ position.  I've also checked the settings on the Web UI for the camera hoping I could somehow trigger the relay output from the camera when it's in a certain PTZ position, but no joy.

Any ideas?

### Version

0.13.2

### Frigate config file

```yaml
N/A
```

### Relevant log output

```shell
N/A
```

### FFprobe output from your camera

```shell
N/A
```

### Operating system

HassOS

### Install method

HassOS Addon

### Object Detector

OpenVino

### Network connection

Wired

### Camera make and model

Dahua SD49412T-HN

**Comment by hawkeye217** [comment]
Nothing is available in Frigate natively for this. However, you could write some simple python code to return the position. 

Though you might consider using an MQTT sensor in Home Assistant for this topic: https://docs.frigate.video/integrations/mqtt#frigatecamera_nameptz_autotrackeractive

If your PTZ is currently autotracking an object, the sensor will be ON. You could then set up that sensor as a condition in your automation.

**Reply by jakeycrx** [reply]
Python is a little bit beyond my skillset! I've setup the MQTT sensor though, which should work fine. Thank you!

**Comment by jakeycrx** [comment, marked as answer]
Just in case anybody comes across this in the future, I managed to work out a fairly simple solution using the command line in HA after sifting through some forum posts.

You can get PTZ info via the API:

![image](https://github.com/user-attachments/assets/656627b7-4153-4d05-a20b-a0f63214937c)

On my camera, that returns:

![image](https://github.com/user-attachments/assets/e8b43974-113f-4ba2-830c-8454107eab07)

status.ActionID and status.PresetID are the PTZ preset numbers you set in the Dahua WebUI. The home position of my camera is preset 1, so I can use that to check if the camera is in the home position.

The command I used is:
```
command_line:
  sensor:
    name: "PTZ Preset Position"
    command: curl -s -g --digest -u user:password "http://192.168.1.108/cgi-bin/ptz.cgi?action=getStatus"
    value_template: '{{ (value|regex_findall("status.PresetID=([\d\.]*)"))[0] }}'
```
The sensor returns the numbered presets:

![image](https://github.com/user-attachments/assets/d4e0d798-bd73-43ab-a41e-d572cfde4507)

I've left the default refresh interval at 30 seconds for the sensor as I don't want to spam the camera. I'll just use the refresh entity service when I want to check the position in automations. 

![image](https://github.com/user-attachments/assets/17627717-e407-4ac5-b377-eb2c85729966)
